### PR TITLE
Zoltan2 partition tree phg interface

### PR DESCRIPTION
**Pull request by MicheldeMessieres**
This PR stacks on PR #1007 so it includes it.
The partitionTree.cpp unit test will now run both rcb and phg and collect partition tree data for each.

### CODE REVIEW

File: packages/zoltan2/test/partition/partitioningTree.cpp [hunk 1]
```diff
@@ -216,11 +237,174 @@ int main(int argc, char** argv)
 }
 ////////////////////////////////////////////////////////////////////////////////
 
+
+////////////////////////////////////////////////////////////////////////////////
+// General test function to analyze solution and print out some information
+////////////////////////////////////////////////////////////////////////////////
+bool validate(part_t numTreeVerts,
```
Comment (kddevin) [inline packages/zoltan2/test/partition/partitioningTree.cpp]: Thanks for adding a validation function to the test.

File: packages/zoltan/src/include/zoltan_partition_tree.h [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  int    *right_leaf
+);
+
+/*****************************************************************************/
+/*
+ *  Function to get the PHG tree size.
+ *  Input:
+ *    zz                  --  The Zoltan structure returned by Zoltan_Create.
+ *  Output:
+ *    tree_size           --  will be set to the size of the tree
```
Comment (kddevin) [inline packages/zoltan/src/include/zoltan_partition_tree.h]: What is "tree size"?  Number of nodes?  Number of levels?  Would you please add a clearer description to the comments?  Thanks.

**Comment by kddevin** [comment]
@MicheldeMessieres I checked this PR for interface issues between Zoltan and Zoltan2.  I had only a request for more detailed comments, and thanks for adding a validation method.  

@mmwolf Would you please review for consistency/correctness with your proposed getPartitionTree interface?  If it looks good to you, we can merge.

Thanks, @MicheldeMessieres .

**Comment by mmwolf** [comment]
I looked at this, ran the tests, and this seems fine.  Fairly soon, I plan to also use this functionality in the nested dissection ordering.

File: packages/zoltan/src/include/zoltan_partition_tree.h (hunk 2 above)
Comment (MicheldeMessieres) [inline packages/zoltan/src/include/zoltan_partition_tree.h]: I've added PR #1165 which explains the parameters and conventions more clearly. This PR is just comments (no code changes).

**Pull request merged**